### PR TITLE
feat: replace openjdk with amazoncorretto:17.0.4 on connectors for seсurity compliance

### PR DESCRIPTION
**Pull request by cuyk**
## What
replace openjdk docker base image with amazoncorretto:17.0.4 on connectors for security compliance.

## Note
Failed connectors tests are failing on master as well.

## 🚨 User Impact 🚨
No user impact.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by supertopher** [comment]
closes #2362

File: tools/bin/build_image.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ if [ "$FOLLOW_SYMLINKS" == "true" ]; then
   # to use as the build context
   tar cL "${exclusions[@]}" . | docker build - "${args[@]}"
 else
-  JDK_VERSION="${JDK_VERSION:-17.0.1}"
+  JDK_VERSION="${JDK_VERSION:-17.0.4}"
```
Comment (cpdeethree) [inline tools/bin/build_image.sh:45]: Curious what prompted the version bump here

Comment (cuyk) [reply tools/bin/build_image.sh:45]: The latest [amazoncorretto:17.0.1](https://hub.docker.com/_/amazoncorretto/tags?page=1&name=17.0.1) is 9 months old already and can have some outdated packages.

**Review by evantahler — COMMENTED** [review]
This is great, and we probably should be bumping our base  images every... 6 months?

Some comments:
1. This seems to be only for Java connectors, not the CDK/Python connectors.  For [example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/Dockerfile).  Do those need to be bumped as well?
2. Does this image also work for ARM builds?  Our publish pipeline will also make [MacOS/ARM builds](https://github.com/airbytehq/airbyte/blob/master/tools/integrations/manage.sh#L214).
3. Will this PR re-publish all relevant connectors?

**Comment by cuyk** [comment]
> 1. This seems to be only for Java connectors, not the CDK/Python connectors. For [example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/Dockerfile). Do those need to be bumped as well?
>  2.  Does this image also work for ARM builds? Our publish pipeline will also make [MacOS/ARM builds](https://github.com/airbytehq/airbyte/blob/master/tools/integrations/manage.sh#L214).
>  3.  Will this PR re-publish all relevant connectors?
1. Yeah, it's java connectors only. I filed an [issue](https://github.com/airbytehq/airbyte-cloud/issues/2998) to follow up with python-based connectors.
2. https://hub.docker.com/_/amazoncorretto/tags?page=1&name=17.0.4 has both **amd64** and **arm64**, so it shoud work as it was
3. Nice catch. According to [docs](https://docs.airbyte.com/connector-development/#publishing-a-connector) connector version should be bumped manually.

**Review by grishick — APPROVED** [review]

**Review by davinchia — APPROVED** [review]

**Pull request merged**